zerver/lib/upload.py

```diff
@@ -181,13 +188,27 @@ def resize_emoji(image_data: bytes, size: int = DEFAULT_EMOJI_SIZE) -> bytes:
                 or im.size[0] > MAX_EMOJI_GIF_SIZE  # dimensions too large
                 or len(image_data) > MAX_EMOJI_GIF_FILE_SIZE_BYTES  # filesize too large
             )
-            return resize_gif(im, size) if should_resize else image_data
+
+            # Generating still image.
+            im.seek(1)
+            still_image = im.copy()
+
+            if should_resize:
+                image_data = resize_gif(im, size)
+                still_image = ImageOps.exif_transpose(still_image)
+                still_image = ImageOps.fit(still_image, (size, size), Image.ANTIALIAS)
+
+            out = io.BytesIO()
+            still_image.save(out, format="PNG")
+            still_image_data = out.getvalue()
```

Comment:
The ordering of lines is odd here.  I kinda suspect that converting to PNG is likely to trigger whatever GIF plugin bugs are involved in should_resize=False anyway, so we should probably resize unconditionally here?  Not sure.  

It's worth testing with an example image that where this is relevant and seeing what happens with the resize/no-resize cases.  (I think there's a few examples in https://github.com/zulip/zulip/issues/10351).

Reply:
I changed this with my push, but we need to test to determine which is the right experience.

Reply:
Okay, So I tried adding,

![bear.gif](https://user-images.githubusercontent.com/3491502/47063808-a8f6a680-d191-11e8-9b22-23d0ea6d148a.gif) (It doesn't need to be resized)

And its still image looks alright,
![Screenshot 2021-09-11 at 1 49 25 PM](https://user-images.githubusercontent.com/63820270/132941497-35d9b6a6-c8ff-4045-87a6-37ed8ba0c553.png)

And same with emoji that requires resize,
![raccon](https://user-images.githubusercontent.com/63820270/132941559-4de9c530-fb42-428a-a28a-a4e1b95c8dc7.gif)

Its still image looks alright,
![Screenshot 2021-09-11 at 1 52 20 PM](https://user-images.githubusercontent.com/63820270/132941584-54e34938-bffb-417e-b340-1a078d67a6de.png)

Reply:
So, I guess doing resize no matter what will be alright!